### PR TITLE
add map

### PR DESCRIPTION
**Pull request by marvinRoellecke**
- create new route page "map
- create Map component
- install leaflet and react-leaflet
- render map
- add map icon to navbar
- adapt form submit to convert address into geocoordinates using geoapify
- change db structure to meet leaflet specs for coordinates

### CODE REVIEW

File: pages/new-entry/index.js [hunk 1]
```diff
@@ -19,7 +19,17 @@ export default function NewEntryForm({ startFetching }) {
     startFetching();
   }
 
-  function handleSubmit(event) {
+  async function getCoordinates(address) {
+    const response = await fetch(
```
Comment (onebarloop) [inline pages/new-entry/index.js:23]: GeoApify 🚀

File: components/Map/Map.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    <>
+      <MapContainer
+        center={[51.5, 10.2]}
+        zoom={6}
+        scrollWheelZoom={true}
+        style={{ height: "100%", width: "100%" }}
+      >
+        <TileLayer url="https://tiles.stadiamaps.com/tiles/alidade_smooth/{z}/{x}/{y}{r}.png" />
+
+        {locations.map((location) => (
```
Comment (onebarloop) [inline components/Map/Map.js:23]: Matter of taste: You could destructure location here:

```javascript
{locations.map({id, coordinates, title, info}) => (
...
)}
```

File: db/models/Location.js [hunk 3]
```diff
@@ -17,8 +17,8 @@ const locationSchema = new Schema({
     postcode: { type: Number, required: true },
     city: { type: String, required: true },
   },
-  latitude: { type: String, required: false },
-  longitude: { type: String, required: false },
+  coordinates: [{ type: Number, required: false }],
```
Comment (onebarloop) [inline db/models/Location.js:20]: Does this work properly? My Schema looks like this: 

```javascript
coordinates: {type: [Number], required: false}
```

But maybe both Syntaxes work?!

**Review by onebarloop — COMMENTED** [review]
Nice one Marvin. Just some tiny comments

File: components/Map/Map.js (hunk 2 above)
Comment (marvinRoellecke) [reply components/Map/Map.js:23]: thanks for the hint...I will plan that for the "final" refactor

File: db/models/Location.js (hunk 3 above)
Comment (marvinRoellecke) [reply db/models/Location.js:20]: ups...your version should be the right way. but yes it works 😬

File: pages/new-entry/index.js (hunk 1 above)
Comment (marvinRoellecke) [reply pages/new-entry/index.js:23]: saw that in your code 🤭

**Pull request merged**